src/k8s-engine/lima.ts

```diff
@@ -1554,7 +1554,44 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         ]);
         if (config.enabled) {
           this.lastCommandComment = 'Checking k3s images';
-          await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          try {
+            await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          } catch (ex:any) {
+            console.log(`Failed to find version ${ desiredVersion.raw }: ${ ex }`, ex);
+            if (ex.code === 'ECONNREFUSED' || ex.toString().includes('getaddrinfo ENOTFOUND github.com') || ex.toString().includes('getaddrinfo EAI_AGAIN github.com')) {
```

Comment:
There should be a better way of checking for errors than string comparison — or at least, factor this out into a helper function.

Reply:
Yes, I'm replacing checking errors with a call to `dns.lookup`. It has the good property of failing immediately if there's a network problem.